pandas/tests/indexes/datetimes/test_timezones.py

```diff
@@ -0,0 +1,1047 @@
+        ts = ['2008-05-12 09:50:00',
+              '2008-12-12 09:50:35',
+              '2008-05-12 09:50:32']
+        tt = DatetimeIndex(ts).tz_localize('US/Eastern')
+        ut = tt.tz_convert('UTC')
+        expected = Index([13, 14, 13])
+        tm.assert_index_equal(ut.hour, expected)
+
+        # unsorted case UTC -> US/Eastern
+        ts = ['2008-05-12 13:50:00',
+              '2008-12-12 14:50:35',
+              '2008-05-12 13:50:32']
+        tt = DatetimeIndex(ts).tz_localize('UTC')
+        ut = tt.tz_convert('US/Eastern')
+        expected = Index([9, 9, 9])
+        tm.assert_index_equal(ut.hour, expected)
+
+    @pytest.mark.parametrize('tz', ['US/Eastern', 'dateutil/US/Eastern'])
+    def test_dti_tz_convert_hour_overflow_dst_timestamps(self, tz):
+        # Regression test for GH#13306
+
+        # sorted case US/Eastern -> UTC
+        ts = [Timestamp('2008-05-12 09:50:00', tz=tz),
+              Timestamp('2008-12-12 09:50:35', tz=tz),
+              Timestamp('2009-05-12 09:50:32', tz=tz)]
+        tt = DatetimeIndex(ts)
+        ut = tt.tz_convert('UTC')
+        expected = Index([13, 14, 13])
+        tm.assert_index_equal(ut.hour, expected)
+
+        # sorted case UTC -> US/Eastern
+        ts = [Timestamp('2008-05-12 13:50:00', tz='UTC'),
+              Timestamp('2008-12-12 14:50:35', tz='UTC'),
+              Timestamp('2009-05-12 13:50:32', tz='UTC')]
+        tt = DatetimeIndex(ts)
+        ut = tt.tz_convert('US/Eastern')
+        expected = Index([9, 9, 9])
+        tm.assert_index_equal(ut.hour, expected)
+
+        # unsorted case US/Eastern -> UTC
+        ts = [Timestamp('2008-05-12 09:50:00', tz=tz),
+              Timestamp('2008-12-12 09:50:35', tz=tz),
+              Timestamp('2008-05-12 09:50:32', tz=tz)]
+        tt = DatetimeIndex(ts)
+        ut = tt.tz_convert('UTC')
+        expected = Index([13, 14, 13])
+        tm.assert_index_equal(ut.hour, expected)
+
+        # unsorted case UTC -> US/Eastern
+        ts = [Timestamp('2008-05-12 13:50:00', tz='UTC'),
+              Timestamp('2008-12-12 14:50:35', tz='UTC'),
+              Timestamp('2008-05-12 13:50:32', tz='UTC')]
+        tt = DatetimeIndex(ts)
+        ut = tt.tz_convert('US/Eastern')
+        expected = Index([9, 9, 9])
+        tm.assert_index_equal(ut.hour, expected)
+
+    def test_dti_tz_convert_trans_pos_plus_1__bug(self):
+        # Regression test for tslib.tz_convert(vals, tz1, tz2).
+        # See https://github.com/pandas-dev/pandas/issues/4496 for details.
+        for freq, n in [('H', 1), ('T', 60), ('S', 3600)]:
+            idx = date_range(datetime(2011, 3, 26, 23),
+                             datetime(2011, 3, 27, 1), freq=freq)
+            idx = idx.tz_localize('UTC')
+            idx = idx.tz_convert('Europe/Moscow')
+
+            expected = np.repeat(np.array([3, 4, 5]), np.array([n, n, 1]))
+            tm.assert_index_equal(idx.hour, Index(expected))
+
+    def test_dti_tz_convert_dst(self):
```

Comment:
parameterize (can be next pass)

Reply:
Yah I parametrized the most egregious of these, but there is bunch of cleanup like this for the next pass.